src/screens/rewards/RewardsSheet.tsx

```diff
@@ -17,9 +17,15 @@ export const RewardsSheet: React.FC = () => {
   const accountAddress = useSelector(
     (state: AppState) => state.settings.accountAddress
   );
-  const { data, isLoading } = useRewards({
+  const [isLoading, setIsLoading] = useState(true);
+  const { data, isLoading: queryIsLoading } = useRewards({
     address: accountAddress,
   });
+
+  useEffect(() => {
+    setIsLoading(queryIsLoading);
```

Comment:
is there a reason not to use the query's `isLoading` directly?

Reply:
Yes, I should probably comment that. I explicitly want to show the loading UI, because rendering all the items may take some time on lower end devices. There will be 50 list elements with images etc. And with the react query isLoading, when the content is cached, there's no loading at all.

So that was the reason, I think the UX will be better if we actually show the loading